module/zfs/vdev_queue.c

```diff
@@ -614,7 +614,7 @@ vdev_queue_aggregate(vdev_queue_t *vq, zio_t *zio)
 		limit = zfs_vdev_aggregation_limit_non_rotating;
 	else
 		limit = zfs_vdev_aggregation_limit;
-	limit = MAX(MIN(limit, maxblocksize), 0);
+	limit = MIN(limit, maxblocksize);
```

Comment:
In this case it actually looks like an actual problem or at least a mess, since maxblocksize is signed and values assigned to limit above are also, but both are then converted to uint64_t.

Reply:
We will never have a negative maxblocksize, so we are fine. We could do further cleanup by modifying `spa_maxblocksize()` to only return unsigned values since the constants that it uses for its values are unsigned, but I did not feel strongly enough about this to make that change.

Reply:
zfs_vdev_aggregation_limit_non_rotating and zfs_vdev_aggregation_limit are signed int module parameters.

Reply:
`limit` is unsigned, so I had not considered those two being signed. I guess we should change those two to be unsigned values too. Presumably, bad things could happen if they are made negative.

Reply:
I just opened #13875 to make zfs_vdev_aggregation_limit_non_rotating, zfs_vdev_aggregation_limit and many other things that should never be negative unsigned. I am marking this conversation as resolved.